_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,338 @@
+
+Besides the High Level REST Client support, Spring Data provides the [_Reactive Client_](https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#elasticsearch.clients.reactive), a non-official driver based on WebClient, with calls operated directly on the reactive stack. The Reactive Client also depends on the Elasticsearch core as it is designed for handling Elasicsearch request/response types.
+
+## Build an application with Spring Data Elasticsearch
+
+JHipster provides the Elasticsearch option to add search capabilities on top of your database. The integration is based on Spring Data Elasticsearch repositories, so let's generate a reactive Blog application to explore what the generator provides. The Blog application built in this tutorial is based on the JHipster sample [reactive-ms.jdl](https://github.com/jhipster/jdl-samples/blob/main/reactive-ms.jdl), but adding Maven, MongoDB (Elasticsearch integration only works with SQL databases and MongoDB), Bootstrap pagination, Okta authentication, and Kibana for index mapping inspection.
+
+Start by creating a `spring-data-elasticsearch` folder for the project, fetch the application JDL from the [GitHub repository](https://github.com/oktadev/okta-spring-data-elasticsearch-example), and generate the application with JHipster:
+
+```shell
+mkdir spring-data-elasticsearch
+cd spring-data-elasticsearch
+https -d raw.githubusercontent.com/oktadev/okta-spring-data-elasticsearch-example/main/blog-reactive-ms.jdl
+jhipster jdl blog-reactive-ms.jdl
+```
+
+{% include setup/cli.md type="jhipster"
+   loginRedirectUri="http://localhost:8080/login/oauth2/code/oidc,http://localhost:8081/login/oauth2/code/oidc,http://localhost:8761/login/oauth2/code/oidc"
+   logoutRedirectUri="http://localhost:8080,http://localhost:8081,http://localhost:8761" %}
+
+
+The JHipster Registry is also a Spring Cloud Config server, and by default, it is configured with the profiles `dev` and `native`, which means the configuration will be provided from the location `docker-compose/central-server-config`. Update `docker-compose/central-server-config/application.yml` with the OIDC settings to be shared with all microservices. Set the values from the `.okta.env` file created with Okta CLI.
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://{yourOktaDomain}/oauth2/default
+        registration:
+          oidc:
+            client-id: {clientId}
+            client-secret: {clientSecret}
+```
```

Comment:
Here's a way to do this from the command line. It might help eliminate copy/paste issues.

```
source ~/.okta.env

echo "
spring:
  security:
    oauth2:
      client:
        provider:
          oidc:
            issuer-uri: $SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI
        registration:
          oidc:
            client-id: $SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID
            client-secret: $SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET" 
>> docker-compose/central-server-config/application.yml
```

Reply:
I will add the tip to the post. My .okta.env is created in the root app folder. I did source .okta.env. Please confirm which .okta.env location to use.

Reply:
You are correct. I removed the `~/` from my `source` command.